src/parse_util.rs

```diff
@@ -1505,14 +1536,18 @@
     }
 
     // Get the statement we are part of.
-    let st = dst.parent().unwrap().as_statement().unwrap();
+    let NodeRef::Branch(BranchRef::Statement(st)) = dst.parent().unwrap().as_node() else {
+        panic!()
+    };
 
     // Walk up to the job.
     let mut job = None;
     let mut cursor = dst.parent();
     while job.is_none() {
         let c = cursor.expect("Reached root without finding a job");
-        job = c.as_job_pipeline();
+        if let NodeRef::Branch(BranchRef::JobPipeline(j)) = c.as_node() {
+            job = Some(j);
```

Comment:
fwiw this loop should be rewritten to use `break job;`, then job doesn't need to be mutable